metals/src/main/scala/scala/meta/internal/metals/ScalaTarget.scala

```diff
@@ -26,7 +26,13 @@ case class ScalaTarget(
 
   def targetroot: AbsolutePath = scalac.targetroot(scalaVersion)
 
-  def baseDirectory: String = info.getBaseDirectory()
+  def baseDirectory: String = {
+    val baseDir = info.getBaseDirectory()
+    // In the BSP spec base directory doesn't have to be defined, so we do a
+    // null check here and just give it a "" if it's null. Mainly this just
+    // impacts the order of the build targets in the Doctor and tree view.
+    if (baseDir != null) baseDir else ""
```

Comment:
So this isn't related to sbt, but it seemed relevant so I added it in. I discovered this when I actually wanted to try out a workspace with multiple `.bsp/entries` and I tested the new Mill BSP support, and they had null values for some targets causing this to throw when we tried to sort by it in the Doctor.

Reply:
On second thought, seeing that this isn't related to these changes, I'll just include this in another pr.

https://github.com/scalameta/metals/pull/2164